### PR TITLE
Input type required_with prefix

### PR DESCRIPTION
**Pull request by stevelacey**
## Summary

Input type validation rules need to prefix the field reference in the `required_with` rules, otherwise the rules would have to be pushed up to the query level.

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)

### Checklist
- [ ] Existing tests have been adapted and/or new tests have been added
- [ ] Add a CHANGELOG.md entry
- [x] Code style has been fixed via `composer fix-style`

### CODE REVIEW

**Comment by stevelacey** [comment]
@mfn what do you think about this, shall I finish it up? I figure this needs to work

**Comment by mfn** [comment]
Sorry for missing this. Was pretty sure I already looked at it.

The manual preg_match for a rulename feels strange but I undertsand, there's no other way to make this probably work.

Would like to tie in @crissi who did some great work on validation what he things?

**Comment by crissi** [comment]
> @mfn what do you think about this, shall I finish it up? I figure this needs to work
> 
> **Edit:** I actually have some other rules that I've written that behave like `required_with` and accept a field arg that would continue to not work here, I wonder if there's a more comprehensive solution to the problem; is there a good reason we're building out indexed rules for arrays as opposed a asterisk `x.*.y` rule? I assume this is important and necessary, but maybe it's not. Perhaps validation needs a rework.

 I remember this coming up during some work with rules, but I right now I actually can't see why it wouldn't work with *

**Comment by crissi** [comment]
never worked with the required_with so not much to say here, but it makes sense that we need params from the same depth here.

The PR could use some tests though

File: src/Support/Rules.php [hunk 1]
```diff
@@ -44,12 +44,34 @@ public function get(): array
...
+    protected function resolveRules($rules, ?string $prefix, array $arguments)
     {
         if (is_callable($rules)) {
             return call_user_func($rules, $arguments, $this->requestArguments);
         }
 
+        if ($prefix && is_array($rules)) {
+            $rules = array_map(
+                function ($rule) use ($prefix) {
+                    if (is_string($rule) && preg_match('/^(required_with(?:out)?(?:_all)?):(.*)$/', $rule, $matches)) {
```
Comment (mfn) [inline src/Support/Rules.php:56]: Hm, don't we also support the array forms of rules, e.g. `['required_with:…', '…']` or am I on the completely wrong track here?

Yeah, I think tests to play around would help :)

**Comment by stevelacey** [comment]
I somewhat lost appetite for this approach as soon as I realised that it wouldn't cover my custom `absent_with` extension (the inverse of `required_with`) without additional effort, to do that I'd also need to add a way to tell the package which rule parameters are "field parameters" and should thus be prefixed inside input types... that... or do something like "check if there's a field that === this name?" which might be a sketchy approach.

File: src/Support/Rules.php (hunk 1 above)
Comment (stevelacey) [reply src/Support/Rules.php:56]: Yeah `['required_with' => ['thing']]` might be a thing

Comment (stevelacey) [reply src/Support/Rules.php:56]: Also this is apparently a thing

```php
                'rules' => function ($rules) {
                    return ['required'];
                },
```

Not sure how viable this approach is.

**Comment by mfn** [comment]
> I somewhat lost appetite for this approach as

Shall we close the PR then 🤔?

**Comment by stevelacey** [comment]
Yep, I went with query-level array rules (`thing.*.thing`) in the end, it's awkward but saved addressing this problem.

**Pull request closed without merging**